include/nbl/builtin/hlsl/memory_accessor.hlsl

```diff
@@ -16,73 +16,43 @@ struct MemoryAdaptor
+    void get(const uint ix, NBL_REF_ARG(vector <Scalar, 4>) value) { accessor.get(ix, value.x), accessor.get(ix + _NBL_HLSL_WORKGROUP_SIZE_, value.y), accessor.get(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, value.z), accessor.get(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_, value.w);}
 
-    void set(const uint ix, const uint value) {accessor.set(ix, value);}
-    void set(const uint ix, const uint2 value) {
+    template<typename Scalar>
+    void set(const uint ix, const Scalar value) {accessor.set(ix, value);}
+    template<typename Scalar>    
+    void set(const uint ix, const vector <Scalar, 2> value) {
         accessor.set(ix, value.x);
         accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, value.y);
     }
-    void set(const uint ix, const uint3 value)  {
+    template<typename Scalar>
+    void set(const uint ix, const <Scalar, 3> value)  {
         accessor.set(ix, value.x);
         accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, value.y);
         accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, value.z);
     }
-    void set(const uint ix, const uint4 value) {
+    template<typename Scalar>
+    void set(const uint ix, const <Scalar, 4> value) {
         accessor.set(ix, value.x);
         accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, value.y);
         accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, value.z);
         accessor.set(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_, value.w);
     }
-
-    void set(const uint ix, const int value) {accessor.set(ix, asuint(value));}
-    void set(const uint ix, const int2 value) {
-        accessor.set(ix, asuint(value.x));
-        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
-    }
-    void set(const uint ix, const int3 value)  {
-        accessor.set(ix, asuint(value.x));
-        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
-        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
-    }
-    void set(const uint ix, const int4 value) {
-        accessor.set(ix, asuint(value.x));
-        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
-        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
-        accessor.set(ix + 3 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.w));
-    }
-
-    void set(const uint ix, const float value) {accessor.set(ix, asuint(value));}
-    void set(const uint ix, const float2 value) {
-        accessor.set(ix, asuint(value.x));
-        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
-    }
-    void set(const uint ix, const float3 value)  {
-        accessor.set(ix, asuint(value.x));
-        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
-        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
-    }
-    void set(const uint ix, const float4 value) {
-        accessor.set(ix, asuint(value.x));
-        accessor.set(ix + _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.y));
-        accessor.set(ix + 2 * _NBL_HLSL_WORKGROUP_SIZE_, asuint(value.z));
```

Comment:
again, same reason as the other comments, `asuint` was crucial in the functioning of the adaptor